pypesto/optimize/task.py

```diff
@@ -53,6 +53,8 @@ def __init__(
     def execute(self) -> OptimizerResult:
         """Execute the task."""
         logger.debug(f"Executing task {self.id}.")
+        # check for supplied x_guess support
+        self.optimizer.check_x0_support(self.problem)
```

Comment:
could also directly pass self.problem.x_guesses; would be cleaner?

Reply:
but x_guesses is not none if we initialised the problem but `np.zeros((0, self.dim_full))` which is an empty array. Could probably check for that 🤔